pkg/controller/useraccount/useraccount_controller.go

```diff
@@ -213,6 +230,69 @@ func (r *ReconcileUserAccount) ensureIdentity(logger logr.Logger, userAcc *toolc
 	return identity, false, nil
 }
 
+// Sets the finalizers for UserAccount
+func (r *ReconcileUserAccount) setFinalizers(userAcc *toolchainv1alpha1.UserAccount) error {
+	// Add the finalizer if it is not present
+	if !containsString(userAcc.ObjectMeta.Finalizers, userAccFinalizerName) {
+		userAcc.ObjectMeta.Finalizers = append(userAcc.ObjectMeta.Finalizers, userAccFinalizerName)
+		if err := r.client.Update(context.TODO(), userAcc); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// Handles finalizer logic for UserAccount
+func (r *ReconcileUserAccount) finalizer(logger logr.Logger, userAcc *toolchainv1alpha1.UserAccount) (bool, error) {
```

Comment:
Could we have a comment here describing what the `bool` return value means?

Reply:
Thanks @sbryzak! Fixed here: d744e71b1270e079f720db29090851e268f8585d